awswrangler/s3/_read_parquet.py

```diff
@@ -236,457 +216,269 @@ def _read_parquet_metadata(
+) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+    if config.distributed:
+        dataset = read_datasource(
+            datasource=ParquetDatasource(),
+            parallelism=parallelism,
+            use_threads=use_threads,
+            paths=paths,
+            schema=schema,
             columns=columns,
-            categories=categories,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            path_root=path_root,
+        )
+        return _to_modin(dataset=dataset, kwargs=arrow_kwargs)
+
+    if chunked:
+        return _read_parquet_chunked(
             boto3_session=boto3_session,
-            s3_additional_kwargs=s3_additional_kwargs,
+            paths=paths,
+            path_root=path_root,
+            columns=columns,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            chunked=chunked,
             use_threads=use_threads,
-            version_id=version_id,
-            pyarrow_additional_kwargs=pyarrow_args,
-        ),
-        categories=categories,
-        safe=safe,
-        map_types=map_types,
-        use_threads=use_threads,
-        dataset=dataset,
-        path=path,
-        path_root=path_root,
-        timestamp_as_object=pyarrow_args["timestamp_as_object"],
+            s3_additional_kwargs=s3_additional_kwargs,
+            arrow_kwargs=arrow_kwargs,
+            version_ids=version_ids,
+        )
+
+    executor = _get_executor(use_threads=use_threads)
+    tables = executor.map(
+        _read_parquet_file,
+        boto3_session,
+        paths,
+        itertools.repeat(path_root),
+        itertools.repeat(columns),
+        itertools.repeat(coerce_int96_timestamp_unit),
+        itertools.repeat(s3_additional_kwargs),
+        itertools.repeat(use_threads),
+        [version_ids.get(p) if isinstance(version_ids, dict) else None for p in paths],
     )
-    if validate_schema and columns:
-        for column in columns:
-            if column not in df.columns:
-                raise exceptions.InvalidArgument(f"column: {column} does not exist")
-    return df
+    return _utils.table_refs_to_df(tables=tables, kwargs=arrow_kwargs)
 
 
 def read_parquet(
     path: Union[str, List[str]],
     path_root: Optional[str] = None,
+    dataset: bool = False,
     path_suffix: Union[str, List[str], None] = None,
     path_ignore_suffix: Union[str, List[str], None] = None,
-    version_id: Optional[Union[str, Dict[str, str]]] = None,
     ignore_empty: bool = True,
-    ignore_index: Optional[bool] = None,
     partition_filter: Optional[Callable[[Dict[str, str]], bool]] = None,
     columns: Optional[List[str]] = None,
     validate_schema: bool = False,
-    chunked: Union[bool, int] = False,
-    dataset: bool = False,
-    categories: Optional[List[str]] = None,
-    safe: bool = True,
-    map_types: bool = True,
-    use_threads: Union[bool, int] = True,
+    coerce_int96_timestamp_unit: Optional[str] = None,
```

Comment:
Could this one also be removed and passed through the `pyarrow_kwargs` escape hatch?

Reply:
I am afraid not. Items passed via `arrow_additional_kwargs` are used for the [`to_pandas`](https://arrow.apache.org/docs/6.0/python/generated/pyarrow.Table.html#pyarrow.Table.to_pandas) call. `coerce_int96_timestamp_unit` is not an argument there as you can see

In the current implementation, we use these ugly logics ([1](https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/s3/_read_parquet.py#L1120)/[2](https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/s3/_read_parquet.py#L77)) to handle it. I think being explicit about the argument is much simpler